docs/quick_tutorial/authorization.rst

```diff
@@ -87,8 +92,10 @@ This simple tutorial step can be boiled down to the following:
 - This ACL says that the ``edit`` permission is available on ``Root``  to the
   ``group:editors`` *principal*.
 
-- The registered ``groupfinder`` answers whether a particular user (``editor``)
-  has a particular group (``group:editors``).
+- The ``SecurityPolicy.effective_principals`` method answers whether a particular user (``editor``) is a member of a particular group (``group:editors``).
```

Comment:
Would it be bad to inline the code from `effective_principals` inside the `permits` method?
It is not an interface method anymore, just an internal helper for this specific security policy (similar to `load_identity` with RequestLocalCache); I feel that not having a function named `effective_principals` here would help avoiding confusion now that the whole concept is deprecated (at a framework level anyway, a project could still use it).

Reply:
I thought about this but I think I'd prefer to leave it as is as the code is cleaner and it's only an issue for someone like you who already knows how Pyramid's system works. Also I don't want to stop people from capturing a ref to the policy somewhere and invoking other methods on it like this one if they need to.

Reply:
The only change I'd consider here is pushing the principals into the authenticated_identity and pulling them out of there. This would require constructing a custom identity object instead of using the `User`. For example something like a dict with `user` and `effective_principals` keys. At least then `request.user` wouldn't just be an alias for the identity if I did that.

I'm still leaning toward keeping it the way it is though.

Reply:
Using dicts or a named tuple for users / identity objects make sense to me!

Reply:
Sorry I thought about it again and am gonna leave it as is.